client/client.go

```diff
@@ -278,8 +287,18 @@ func NewSubstreamsClient(config *SubstreamsClientConfig) (cli pbsubstreamsrpc.St
 		}
 	}
 
-	zlog.Debug("creating new client", zap.String("endpoint", endpoint))
-	cli = pbsubstreamsrpc.NewStreamClient(conn)
+	return
+}
+
+func NewSubstreamsClients(config *SubstreamsClientConfig) (cliV2 pbsubstreamsrpcv2.StreamClient, cliV3 pbsubstreamsrpcv3.StreamClient, closeFunc func() error, callOpts []grpc.CallOption, headers Headers, err error) {
```

Comment:
Should it be `NewSubstreamsClientConn(...) ...` and caller would be responsible of :

```
cliV2 := pbsubstreamsrpcv2.NewStreamClient(conn)
cliV3 := pbsubstreamsrpcv3.NewStreamClient(conn)
```